components/edit_channel_purpose_modal/edit_channel_purpose_modal.tsx

```diff
@@ -1,59 +1,48 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ChangeEvent, KeyboardEvent} from 'react';
 import {Modal} from 'react-bootstrap';
-import {FormattedMessage, injectIntl} from 'react-intl';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
+
+import {Channel} from 'mattermost-redux/types/channels';
+import {ActionResult} from 'mattermost-redux/types/actions';
 
 import Constants from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 
-class EditChannelPurposeModal extends React.PureComponent {
-    static propTypes = {
-
-        /*
-         * callback to call when modal will hide
-         */
-        onHide: PropTypes.func.isRequired,
-
-        /*
-         * Channel info object
-         */
-        channel: PropTypes.object,
-
-        /*
-         * Check should we send purpose on CTRL + ENTER
-         */
-        ctrlSend: PropTypes.bool.isRequired,
-
-        intl: PropTypes.any,
-
-        /*
-         * Object with redux action creators
-         */
-        actions: PropTypes.shape({
-
-            /*
-             * Action creator to patch current channel
-             */
-            patchChannel: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Actions = {
+    patchChannel: (channelId: string, patch: Channel) => Promise<ActionResult>;
+}
+type Props = {
+    onHide: () => void;
+    channel?: Channel;
+    ctrlSend: boolean;
+    actions: Actions;
+    intl: IntlShape;
+}
+type State = {
+    purpose: string;
+    serverError: string;
+    show: boolean;
+    submitted: boolean;
+    requestStarted: boolean;
+}
```

Comment:
Nit, should we add blank lines between the types definition?
```suggestion
type Actions = {
    patchChannel: (channelId: string, patch: Channel) => Promise<ActionResult>;
}

type Props = {
    onHide: () => void;
    channel?: Channel;
    ctrlSend: boolean;
    actions: Actions;
    intl: IntlShape;
}

type State = {
    purpose: string;
    serverError: string;
    show: boolean;
    submitted: boolean;
    requestStarted: boolean;
}
```